src/shared/Core/Authentication/OAuth/OAuth2Client.cs

```diff
@@ -104,6 +104,9 @@ public IOAuth2CodeGenerator CodeGenerator
         public async Task<OAuth2AuthorizationCodeResult> GetAuthorizationCodeAsync(IEnumerable<string> scopes,
             IOAuth2WebBrowser browser, IDictionary<string, string> extraQueryParams, CancellationToken ct)
         {
+            var label = "get auth code";
+            using IDisposable region = _trace2.CreateRegion(OAuth2Constants.Trace2Category, label);

```

Comment:
The `GetAuthorizationCodeAsync` method (and the `GetTokenByDeviceCodeAsync` method below) involve a period of time where we wait for user interaction. The former being the user interacting with the launched browser to sign in, and the latter being the user navigating to a login page and entering a device code on another device (and we are polling for the server to say "done").

I wonder if we should scope these regions to only parts that don't include user interaction, else these figures will vary widely and be of no use performance tracing-wise.

Reply:
That's a good point. Since there isn't server interaction without user interaction in these cases, I will just remove these regions.

Reply:
Ahh. I think you may have removed the region from the wrong side of the Device Code flow.

Device code flow consists of two parts:

1. `GetDeviceCodeAsync` which has no user input - it tells the server to start a device code flow operation and prints the code to the user. It should return quickly.

2. `GetTokenByDeviceCodeAsync` which polls the server until the user has completed the sign in on another device. This is the method that is bounded by the user interaction speed.

Looks like you've removed the region from 1 and not 2.